File: app/src/main/java/com/sppe/shrimppaste/data/PhotoDao.java
package com.sppe.shrimppaste.data;

import android.content.Context;

import com.j256.ormlite.dao.Dao;

import java.sql.SQLException;
import java.util.List;

/**
 * Created by WHF on 2017/11/5.
 */

public class PhotoDao {

    private Dao<PhotoEntry,?> dao;

    public PhotoDao(Context context){
        dao = DatabaseHelp.getInstance(context).getDao(PhotoEntry.class);
    }


    public void addPhotoEntryList(PhotoEntry photo) {
        try {
            dao.create(photo);
        } catch (SQLException e) {
            e.printStackTrace();
        }
    }

    public List<PhotoEntry> queryPhotoEntryList(){
        try {
            return dao.queryForAll();
        } catch (SQLException e) {
            e.printStackTrace();
            return null;
        }
    }
}

File: app/src/main/java/com/sppe/shrimppaste/data/PhotoDao.kt
package com.sppe.shrimppaste.data

import android.content.Context
import android.util.Log
import com.j256.ormlite.dao.Dao
import java.sql.SQLException

/**
 * Created by @author WangHaoFei on 2017/11/3.
 */

class PhotoDao(val context: Context) {

    val dao: Dao<PhotoEntry, Int> = DataBaseHelper.getInstance(context).getDaoForClass(PhotoEntry::class.java)

    fun addPhotoEntryList(photo: PhotoEntry) {
        try {
            dao.create(photo)
        } catch (e: SQLException) {
            e.printStackTrace()
        }
    }

    fun queryPhotoEntryList(): List<PhotoEntry> {
        val list = dao.queryForAll()
        return list
    }
}
